crates/accelerate/src/sabre_swap/mod.rs

```diff
@@ -57,10 +60,80 @@ pub enum Heuristic {
     Decay,
 }
 
-struct TrialResult {
-    out_map: HashMap<usize, Vec<[usize; 2]>>,
-    gate_order: Vec<usize>,
-    layout: NLayout,
+/// A container for Sabre mapping results.
+#[pyclass(module = "qiskit._accelerate.sabre_swap")]
+#[derive(Clone, Debug)]
+pub struct SabreResult {
+    #[pyo3(get)]
+    pub map: SwapMap,
+    pub node_order: Vec<usize>,
+    #[pyo3(get)]
+    pub node_block_results: NodeBlockResults,
+}
+
+#[pymethods]
+impl SabreResult {
+    #[getter]
+    fn node_order(&self, py: Python) -> PyObject {
+        self.node_order.to_pyarray(py).into()
```

Comment:
We actually probably want to use `into_pyarray` here if we can. `to_pyarray` will copy the vec into the python heap in numpy owned memory. But the real performance advantage for using a numpy array return here is to have numpy create an array wrapper around a pointer to the rust heap. The `IntoPyArray` trait does this: https://docs.rs/numpy/latest/numpy/convert/trait.IntoPyArray.html#tymethod.into_pyarray

Although I could see the compiler complaining about ownership and lifetimes with this existing as a  method.

Reply:
We can copy the array on the Rust side and then use `into_pyarray` if that's an improvement. We can't use `into_pyarray` otherwise since it'll consume `self.node_order`, right?

Reply:
Yeah, that's right. Not sure what I was thinking the other day. `to_pyarray` is fine here, it's doing the same basic copy in rust. I think I just trigger my memory when I first learned the difference between `to_pyarray` and `into_pyarray` and saw the performance improvement from using `into_pyarray` by skipping the copy.